runtime/lua/vim/lsp/util.lua

```diff
@@ -481,6 +481,29 @@ function M.jump_to_location(location)
   return true
 end
 
+--- Preview a location in a floating windows
+---
+--- behavior depends on type of location:
+---   - for Location, range is shown (e.g., function definition)
+---   - for LocationLink, targetRange is shown (e.g., body of function definition)
+---   - for an array of either, the first element is shown
+---
+--@param location Location, LocationLink, or array thereof
+--@return bufnr,winnr buffer and window number of floating window
+function M.preview_location(location)
+  -- location may be LocationLink or Location (more useful for the former)
+  local uri = location.targetUri or location.uri
+  if uri == nil then return end
+  local bufnr = vim.uri_to_bufnr(uri)
+  if not api.nvim_buf_is_loaded(bufnr) then
+    vim.fn.bufload(bufnr)
+  end
+  local range = location.targetRange or location.range
+  local contents = api.nvim_buf_get_lines(bufnr, range.start.line, range["end"].line+1, false)
```

Comment:
Should we add one to the end of line?

Reply:
What do you mean? Whether it should be `range["end"].line+1` here vs. just `range["end"].line`? (In which case, yes, the `+1` is necessary.)

Reply:
I see why that is necessary :)

Reply:
Then I misunderstood your first comment :) Can you explain it to me?